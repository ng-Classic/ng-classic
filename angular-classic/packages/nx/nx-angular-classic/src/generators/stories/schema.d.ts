export interface StoriesGeneratorOptions {
  name: string;
  interactionTests?: boolean;
  skipFormat?: boolean;
  ignorePaths?: string[];
  /**
   * @deprecated Use interactionTests instead. This option will be removed in v18.
   */
  cypressProject?: string;
  /**
   * @deprecated Use interactionTests instead. This option will be removed in v18.
   */
  generateCypressSpecs?: boolean;
}
