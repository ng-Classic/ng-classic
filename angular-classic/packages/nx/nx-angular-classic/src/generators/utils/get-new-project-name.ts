import { normalizePath } from '@nx/devkit';

/**
 * Joins path segments replacing slashes with dashes
 *
 * @param path
 */
export function getNewProjectName(path: string): string {
  // strip leading '/' or './' or '../' and trailing '/' and replaces '/' with '-'
  return normalizePath(path)
    .replace(/(^\.{0,2}\/|\.{1,2}\/|\/$)/g, '')
    .split('/')
    .filter((x) => !!x)
    .join('-');
}
