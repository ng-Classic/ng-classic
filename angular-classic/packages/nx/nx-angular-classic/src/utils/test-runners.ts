export enum UnitTestRunner {
  Jest = 'jest',
  None = 'none',
}

export enum E2eTestRunner {
  Cypress = 'cypress',
  None = 'none',
}
