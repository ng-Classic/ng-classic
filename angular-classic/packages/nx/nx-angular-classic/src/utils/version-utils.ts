import { coerce, major } from 'semver';
import type {
  PackageVersionNames,
  PackageVersions,
} from './backward-compatible-versions';
import { backwardCompatibleVersions } from './backward-compatible-versions';
import * as versions from './versions';
import { angularVersion } from './versions';

export function getPkgVersionForAngularMajorVersion(
  pkgVersionName: PackageVersionNames,
  angularMajorVersion: number
): string {
  return angularMajorVersion < major(coerce(angularVersion))
    ? backwardCompatibleVersions[`angularV${angularMajorVersion}`]?.[
        pkgVersionName
      ] ?? versions[pkgVersionName]
    : versions[pkgVersionName];
}

export function getPkgVersionsForAngularMajorVersion(
  angularMajorVersion: number
): PackageVersions {
  return angularMajorVersion < major(coerce(angularVersion))
    ? backwardCompatibleVersions[`angularV${angularMajorVersion}`] ?? versions
    : versions;
}
