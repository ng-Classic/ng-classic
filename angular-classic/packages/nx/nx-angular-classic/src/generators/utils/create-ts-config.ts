import type { Tree } from '@nx/devkit';
import { writeJson } from '@nx/devkit';
import { tsConfigBaseOptions } from '@nx/js';
import { getInstalledAngularMajorVersion } from './version-utils';

export { extractTsConfigBase } from '@nx/js';

export function createTsConfig(
  host: Tree,
  projectRoot: string,
  type: 'app' | 'lib',
  options: {
    strict?: boolean;
    style?: string;
    bundler?: string;
    rootProject?: boolean;
  },
  relativePathToRootTsConfig: string
) {
  const majorAngularVersion = getInstalledAngularMajorVersion(host);

  const json = {
    compilerOptions: {
      target: majorAngularVersion === 14 ? 'es2020' : 'es2022',
      useDefineForClassFields: false,
    },
    files: [],
    include: [],
    references: [
      {
        path: type === 'app' ? './tsconfig.app.json' : './tsconfig.lib.json',
      },
    ],
  } as any;

  // inline tsconfig.base.json into the project
  if (options.rootProject) {
    json.compileOnSave = false;
    json.compilerOptions = { ...tsConfigBaseOptions, ...json.compilerOptions };
    json.exclude = ['node_modules', 'tmp'];
  } else {
    json.extends = relativePathToRootTsConfig;
  }

  writeJson(host, `${projectRoot}/tsconfig.json`, json);
}
