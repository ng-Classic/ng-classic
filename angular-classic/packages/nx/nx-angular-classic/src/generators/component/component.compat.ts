import { convertNxGenerator } from '@nx/devkit';
import { warnForSchematicUsage } from '../utils/warn-for-schematic-usage';
import { componentGenerator } from './component';

export default warnForSchematicUsage(convertNxGenerator(componentGenerator));
