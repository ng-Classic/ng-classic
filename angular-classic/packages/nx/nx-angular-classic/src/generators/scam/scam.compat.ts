import scamGenerator from './scam';
import { warnForSchematicUsage } from '../utils/warn-for-schematic-usage';
import { convertNxGenerator } from '@nx/devkit';

export default warnForSchematicUsage(convertNxGenerator(scamGenerator));
