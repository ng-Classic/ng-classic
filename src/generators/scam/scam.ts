import type { Tree } from '@nx/devkit';
import { formatFiles } from '@nx/devkit';
import { componentGenerator } from '../component/component';
import { exportScam } from '../utils/export-scam';
import {
  convertComponentToScam,
  normalizeOptions,
  validateOptions,
} from './lib';
import type { Schema } from './schema';

export async function scamGenerator(tree: Tree, rawOptions: Schema) {
  validateOptions(tree, rawOptions);

  const { inlineScam, ...generatorOptions } = rawOptions;
  await componentGenerator(tree, {
    ...generatorOptions,
    skipImport: true,
    export: false,
    standalone: false,
    skipFormat: true,
  });

  const options = normalizeOptions(tree, rawOptions);
  convertComponentToScam(tree, options);
  exportScam(tree, options);

  if (!options.skipFormat) {
    await formatFiles(tree);
  }
}

export default scamGenerator;
