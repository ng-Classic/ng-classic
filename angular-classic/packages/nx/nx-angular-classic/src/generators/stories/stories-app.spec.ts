import { installedCypressVersion } from '@nx/cypress/src/utils/cypress-version';
import type { Tree } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { componentGenerator } from '../component/component';
import { scamGenerator } from '../scam/scam';
import { generateTestApplication } from '../utils/testing';
import { angularStoriesGenerator } from './stories';

// need to mock cypress otherwise it'll use the nx installed version from package.json
//  which is v9 while we are testing for the new v10 version
jest.mock('@nx/cypress/src/utils/cypress-version');

describe('angularStories generator: applications', () => {
  let tree: Tree;
  const appName = 'test-app';
  let mockedInstalledCypressVersion: jest.Mock<
    ReturnType<typeof installedCypressVersion>
  > = installedCypressVersion as never;

  beforeEach(async () => {
    mockedInstalledCypressVersion.mockReturnValue(10);
    tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    await generateTestApplication(tree, {
      name: appName,
    });
  });

  it('should generate stories file', async () => {
    await angularStoriesGenerator(tree, { name: appName });

    expect(
      tree.exists(`apps/${appName}/src/app/app.component.stories.ts`)
    ).toBeTruthy();
  });

  it('should generate stories file for scam component', async () => {
    await scamGenerator(tree, { name: 'my-scam', project: appName });

    await angularStoriesGenerator(tree, { name: appName });

    expect(
      tree.exists(
        `apps/${appName}/src/app/my-scam/my-scam.component.stories.ts`
      )
    ).toBeTruthy();
  });

  it('should ignore paths', async () => {
    await scamGenerator(tree, { name: 'my-scam', project: appName });

    await angularStoriesGenerator(tree, {
      name: appName,
      ignorePaths: [`apps/${appName}/src/app/my-scam/**`],
    });

    expect(
      tree.exists(
        `apps/${appName}/src/app/my-scam/my-scam.component.stories.ts`
      )
    ).toBeFalsy();
  });

  it('should ignore paths when full path to component is provided', async () => {
    await scamGenerator(tree, { name: 'my-scam', project: appName });

    await angularStoriesGenerator(tree, {
      name: appName,
      ignorePaths: [`apps/${appName}/src/app/my-scam/my-scam.component.ts`],
    });

    expect(
      tree.exists(
        `apps/${appName}/src/app/my-scam/my-scam.component.stories.ts`
      )
    ).toBeFalsy();
  });

  it('should ignore a path that has a nested component, but still generate nested component stories', async () => {
    await componentGenerator(tree, { name: 'component-a', project: appName });
    await componentGenerator(tree, {
      name: 'component-a/component-b',
      project: appName,
    });

    await angularStoriesGenerator(tree, {
      name: appName,
      ignorePaths: [
        `apps/${appName}/src/app/component-a/component-a.component.ts`,
      ],
    });

    expect(
      tree
        .read(
          `apps/${appName}/src/app/component-a/component-b/component-b.component.stories.ts`
        )
        .toString()
    ).toMatchSnapshot();
    expect(
      tree.exists(
        `apps/${appName}/src/app/component-a/component-a.component.stories.ts`
      )
    ).toBeFalsy();
  });

  it('should generate stories file for inline scam component', async () => {
    await scamGenerator(tree, {
      name: 'my-scam',
      project: appName,
      inlineScam: true,
    });

    await angularStoriesGenerator(tree, { name: appName });

    expect(
      tree
        .read(`apps/${appName}/src/app/my-scam/my-scam.component.stories.ts`)
        .toString()
    ).toMatchSnapshot();
  });

  it('should generate cypress spec file', async () => {
    await angularStoriesGenerator(tree, {
      name: appName,
      generateCypressSpecs: true,
    });

    expect(
      tree.exists(`apps/${appName}-e2e/src/e2e/app.component.cy.ts`)
    ).toBeTruthy();
  });
});
