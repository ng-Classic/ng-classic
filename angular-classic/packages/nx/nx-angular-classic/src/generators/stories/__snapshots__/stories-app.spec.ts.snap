// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`angularStories generator: applications should generate stories file for inline scam component 1`] = `
"import { Meta } from '@storybook/angular';
import { MyScamComponent } from './my-scam.component';

export default {
  title: 'MyScamComponent',
  component: MyScamComponent,
} as Meta<MyScamComponent>;

export const Primary = {
  render: (args: MyScamComponent) => ({
    props: args,
  }),
  args: {},
};
"
`;

exports[`angularStories generator: applications should ignore a path that has a nested component, but still generate nested component stories 1`] = `
"import { Meta } from '@storybook/angular';
import { ComponentBComponent } from './component-b.component';

export default {
  title: 'ComponentBComponent',
  component: ComponentBComponent,
} as Meta<ComponentBComponent>;

export const Primary = {
  render: (args: ComponentBComponent) => ({
    props: args,
  }),
  args: {},
};
"
`;
